egs/wsj/s5/steps/dict/learn_lexicon.sh

```diff
@@ -0,0 +1,405 @@
+  echo "                               # data / target vocabulary. It's optional."
+  echo "  --min-prob <float>           # The cut-off parameter used to select pronunciation candidates from phone"
+  echo "                               # decoding. We remove pronunciations with probabilities less than this value"
+  echo "                               # after normalizing the probs s.t. the max-prob is 1.0 for each word."
+  echo "  --prior-mean                 # Mean of priors (summing up to 1) assigned to three exclusive pronunciation"
+  echo "         <float,float,float>   # source: reference lexicon, g2p, and phone decoding (used in the Bayesian"
+  echo "                               # pronunciation selection procedure). We recommend setting a larger prior"
+  echo "                               # mean for the reference lexicon, e.g. '0.6,0.2,0.2'."
+  echo "  --prior-counts-tot <float>   # Total amount of prior counts we add to all pronunciation candidates of"
+  echo "                               # each word. By timing it with the prior mean of a source, and then dividing"
+  echo "                               # by the number of candidates (for a word) from this source, we get the"
+  echo "                               # prior counts we actually add to each candidate."
+  echo "  --variants-prob-mass <float> # In the Bayesian pronunciation selection procedure, for each word, we"
+  echo "                               # choose candidates (from all three sources) with highest posteriors"
+  echo "                               # until the total prob mass hit this amount."
+  echo "                               # It's used in a similar fashion when we apply G2P."
+  echo "  --variants-prob-mass-ref     # In the Bayesian pronunciation selection procedure, for each word,"
+  echo "                               # after the total prob mass of selected candidates hit variants-prob-mass,"
+  echo "                               # we continue to pick up reference candidates with highest posteriors"
+  echo "                               # until the total prob mass hit this amount (must >= variants-prob-mass)."
+  echo "  --affix                      # the affix we want to append to the dir to put the retrained model "
+  echo "                               # and all intermediate outputs, like 'lex'."
+  echo "  --num-gauss                  # number of gaussians for the re-trained SAT model (on top of <src-mdl>)."            
+  echo "  --num-leaves                 # number of leaves for the re-trained SAT model (on top of <src-mdl>)." 
+  echo "  --retrain-src-mdl            # whether we want to re-train the src_mdl (If the ref_dict is large enough,"
+  echo "                               # we may not want to retrain it.)"
+  exit 1
+fi
+
+echo "$0 $@"  # Print the command line for logging
+
+ref_dict=$1
+target_vocab=$2
+data=$3
+src_mdl=$4
+ref_lang=$5
+dest_dict=$6
+dir=${src_mdl}_${affix}_work # Most outputs will be put here. 
```

Comment:
actually if we set dir outside the script, I think we can just remove the variable $affix (default to "lex", which now appears only in the dir name where put the re-trained model), and we can just put the re-trained model under $dir/${src_mdl}_retrained. Is it good?